swabbie-core/src/test/kotlin/com/netflix/spinnaker/swabbie/ResourceTypeHandlerTest.kt

```diff
@@ -155,6 +160,103 @@ object ResourceTypeHandlerTest {
     }
   }
 
+  @Test
+  fun `should get violations`() {
+    val workConfiguration = WorkConfigurationTestHelper.generateWorkConfiguration()
+    val resource1 = createTestResource(resourceId = "1", createTs = createTimestampTenDaysAgo)
+    val resource2 = createTestResource(resourceId = "2", createTs = createTimestampTenDaysAgo)
+    val violationSummary = Summary(description = "test rule", ruleName = "testRule")
+
+    val testRule = TestRule(
+      invalidOn = { it.resourceId == resource1.resourceId },
+      summary = violationSummary
+    )
+
+    defaultHandler.setCandidates(mutableListOf(resource1, resource2))
+    defaultHandler.setRules(mutableListOf(testRule))
+
+    expectThat(
+      defaultHandler.getViolations(resource1, workConfiguration = workConfiguration)
+    ).isEqualTo(
+      listOf(violationSummary)
+    )
+
+    expectThat(
+      defaultHandler.getViolations(resource2, workConfiguration = workConfiguration)
+    ).isEmpty()
+  }
+
+  @Test
+  fun `should get violations with enabled rules`() {
+    val workConfiguration = WorkConfigurationTestHelper.generateWorkConfiguration()
+    val resource = createTestResource(resourceId = "1", createTs = createTimestampTenDaysAgo)
+    val testRule1 = TestRule(
+      name = "testRule1",
+      invalidOn = { it.resourceId == resource.resourceId },
+      summary = Summary(description = "test rule", ruleName = "testRule1")
+    )
+
+    val testRule2 = TestRule(
+      name = "testRule2",
+      invalidOn = { it.resourceId == "other" },
+      summary = Summary(description = "test rule", ruleName = "testRule2")
+    )
+
+    // applies if all enabled rules apply
+    val andRuleConfig = RuleConfiguration()
+      .apply {
+        operator = AND
+        rules = listOf(testRule1.name(), testRule2.name())
+      }
+
+    // applies if any of the enabled rules apply
+    val orRuleConfig = RuleConfiguration()
+      .apply {
+        operator = OR
+        rules = listOf(testRule1.name(), testRule2.name())
+      }
+
+    val orNonApplyingRuleConfig = RuleConfiguration()
+      .apply {
+        operator = OR
+        rules = listOf(testRule2.name())
+      }
+
+    defaultHandler.setCandidates(mutableListOf(resource))
+    defaultHandler.setRules(mutableListOf(testRule1, testRule2))
+
+    // no enabled rules goes through basic behavior of evaluating the resource against all rules
+    // in this case testRule1 applies to this resource
+    expectThat(
+      defaultHandler.getViolations(resource, workConfiguration.copy(enabledRules = emptyList()))
+    ).isNotEmpty()
+      .get {
+        this.map { it.ruleName }
+      }.isEqualTo(
+        listOf(testRule1.name())
+      )
+
+    // andRuleConfig
```

Comment:
test with `andRule` resulting in violation would be good

Reply:
Good call sir! Adding this test found a line I lost in one of the commits.